### PR TITLE
Several updates to `$join()`

### PR DESCRIPTION
**Pull request by etiennebacher**
See changes in NEWS

### CODE REVIEW

File: R/dataframe__frame.R [hunk 1]
```diff
@@ -919,14 +907,20 @@ DataFrame_to_list = function(unnest_structs = TRUE) {
...
 DataFrame_join = function(
-    other, # : LazyFrame or DataFrame,
-    left_on = NULL, # : str | pli.RPolarsExpr | Sequence[str | pli.RPolarsExpr] | None = None,
-    right_on = NULL, # : str | pli.RPolarsExpr | Sequence[str | pli.RPolarsExpr] | None = None,
-    on = NULL, # : str | pli.RPolarsExpr | Sequence[str | pli.RPolarsExpr] | None = None,
-    how = c("inner", "left", "outer", "semi", "anti", "cross"),
+    other,
+    on = NULL,
+    how = c("inner", "left", "outer", "semi", "anti", "cross", "outer_coalesce"),
+    left_on = NULL,
```
Comment (eitsupi) [inline R/dataframe__frame.R:913]:
```suggestion
    how = c("inner", "left", "outer", "semi", "anti", "cross", "outer_coalesce"),
    ...,
    left_on = NULL,
```

https://github.com/pola-rs/polars/blob/c9a14687acf49019b755d8e2141917766bbd5eae/py-polars/polars/dataframe/frame.py#L6234-L6245

File: R/lazyframe__lazy.R [hunk 2]
```diff
@@ -1070,19 +1101,32 @@ LazyFrame_group_by = function(..., maintain_order = pl$options$maintain_order) {
...
 LazyFrame_join = function(
-    other, # : LazyFrame or DataFrame,
-    left_on = NULL, # : str | pli.RPolarsExpr | Sequence[str | pli.RPolarsExpr] | None = None,
-    right_on = NULL, # : str | pli.RPolarsExpr | Sequence[str | pli.RPolarsExpr] | None = None,
-    on = NULL, # : str | pli.RPolarsExpr | Sequence[str | pli.RPolarsExpr] | None = None,
-    how = c("inner", "left", "outer", "semi", "anti", "cross"),
+    other,
+    on = NULL,
+    how = c("inner", "left", "outer", "semi", "anti", "cross", "outer_coalesce"),
+    left_on = NULL,
```
Comment (eitsupi) [inline R/lazyframe__lazy.R:1119]: Ditto.

File: R/lazyframe__lazy.R [hunk 3]
```diff
@@ -1070,19 +1101,32 @@ LazyFrame_group_by = function(..., maintain_order = pl$options$maintain_order) {
...
+    validate = "m:m",
+    join_nulls = FALSE,
     allow_parallel = TRUE,
     force_parallel = FALSE) {
   uw = \(res) unwrap(res, "in $join():")
 
-  if (inherits(other, "RPolarsDataFrame")) {
-    other = other$lazy()
+  if (!inherits(other, "RPolarsLazyFrame")) {
+    Err_plain("`other` must be a LazyFrame.") |> uw()
```
Comment (eitsupi) [inline R/lazyframe__lazy.R:1129]: - Could you please note this change in the NEWS?
- Could you use `is_polars_lf` instead of `inherits`?

**Review by eitsupi — CHANGES_REQUESTED** [review]
Thanks!

File: R/dataframe__frame.R (hunk 1 above)
Comment (etiennebacher) [reply R/dataframe__frame.R]: Done, thanks

**Review by eitsupi — APPROVED** [review]

**Pull request merged**